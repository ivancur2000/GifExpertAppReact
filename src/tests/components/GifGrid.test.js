import React from 'react'; 
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente GifGrid', ()=>{

    const category = 'One Punch';
    
    test('el componete debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe mostrar data del useFetch', ()=>{


        const gifs = [{
            id: '123',
            url: 'http://localhost.imh2',
            title: 'ImageTest',
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={ category }/>);
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

});