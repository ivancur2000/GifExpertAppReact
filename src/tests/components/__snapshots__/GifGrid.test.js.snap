// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid debe mostrar data del useFetch 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX"
  >
    One Punch
  </h3>
  <div
    className="cardGrid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="ImageTest"
      url="http://localhost.imh2"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid el componete debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__flipInX"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="cardGrid"
  />
</Fragment>
`;
